src/openms/source/FORMAT/AbsoluteQuantitationStandardsFile.cpp

```diff
@@ -0,0 +1,70 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2018.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/FORMAT/AbsoluteQuantitationStandardsFile.h>
+
+namespace OpenMS
+{
+  AbsoluteQuantitationStandardsFile::AbsoluteQuantitationStandardsFile() {}
+
+  AbsoluteQuantitationStandardsFile::~AbsoluteQuantitationStandardsFile() {}
+
+  void AbsoluteQuantitationStandardsFile::load(
+    const String& filename,
+    std::vector<AbsoluteQuantitationStandards::runConcentration>& run_concentrations
+  ) const
+  {
+    CsvFile csv(filename);
+    StringList sl;
+    std::map<String, Size> headers_i;
+    csv.getRow(0, sl);
+    for (Size i = 0; i < sl.size(); ++i)
+    {
+      headers_i[sl[i]] = i;
+    }
+    for (Size i = 1; i < csv.rowCount(); ++i)
```

Comment:
It maybe better to implement this at two different methods (e.g., `parseHeader` and `parseLine`).  See https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/FORMAT/AbsoluteQuantitationMethodFile.cpp for what I did previously.  There maybe a simpler way todo it but it is important to take care of default values and also to check for the correct value before trying to cast as e.g., a double.